src/node_crypto.cc

```diff
@@ -741,6 +743,61 @@ void SecureContext::AddCRL(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void UseExtraCaCerts(const char* fname) {
+  extra_root_certs_file = fname;
+}
+
+static void EmitWarning(Environment* env, const char* fmt, ...) {
+  char warning[1024];
+  va_list ap;
+
+  va_start(ap, fmt);
+  vsnprintf(warning, sizeof(warning), fmt, ap);
+  va_end(ap);
+
+  HandleScope handle_scope(env->isolate());
+  Context::Scope context_scope(env->context());
+
+  Local<Object> process = env->process_object();
+  Local<v8::Value> f = process->Get(
+    FIXED_ONE_BYTE_STRING(env->isolate(), "emitWarning"));
+  Local<Value> arg = node::OneByteString(env->isolate(), warning);
+
+  // MakeCallback() unneeded, because emitWarning is internal-code, it calls
+  // process.emit('warning', ..), but does so in the nextTick.
+  f.As<v8::Function>()->Call(process, 1, &arg);
+}
+
+static void LoadExtraCaCerts(Environment* env) {
+  if (!extra_root_certs_file)
+    return;
+
+  MarkPopErrorOnReturn mark_pop_error_on_return;
+
+  BIO* bio = BIO_new_file(extra_root_certs_file, "r");
+  if (!bio) {
+    unsigned long err = ERR_get_error();  // NOLINT(runtime/int)
+    EmitWarning(env, "Ignoring extra certs from `%s`, load failed: %s\n",
+                extra_root_certs_file, ERR_error_string(err, NULL));
+    return;
+  }
+  while (X509* x509 =
+      PEM_read_bio_X509(bio, nullptr, CryptoPemCallback, nullptr)) {
+    X509_STORE_add_cert(root_cert_store, x509);
+    X509_free(x509);
+  }
+  BIO_free_all(bio);
+
+  unsigned long err = ERR_peek_error();  // NOLINT(runtime/int)
+  // Clear error if its EOF/no start line found.
+  if (ERR_GET_LIB(err) == ERR_LIB_PEM &&
+      ERR_GET_REASON(err) == PEM_R_NO_START_LINE) {
+    ERR_clear_error();
```

Comment:
You don't need to call `ERR_clear_error()` if you have a MarkPopErrorOnReturn on the stack.

As well, using `ERR_peek_error()` is not really correct because it an existing error.  I don't really know a good way around that except maybe for calling `ERR_clear_error()` on function entry or checking that `ERR_get_state()->top` changed.